examples/ui-demo/src/hooks/useGetEthPrice.ts

```diff
@@ -0,0 +1,73 @@
+import { useQuery } from "@tanstack/react-query";
+import { createPublicClient, http, formatUnits, parseAbi, Address } from "viem";
+import { mainnet } from "viem/chains";
+
+// Chainlink ETH/USD Price Feed on Ethereum Mainnet
+const ETH_USD_PRICE_FEED_ADDRESS: Address =
+  "0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419";
+
+const aggregatorV3InterfaceABI = parseAbi([
+  "function latestRoundData() external view returns (uint80 roundId, int256 answer, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound)",
+  "function decimals() external view returns (uint8)",
+]);
+
+const publicClient = createPublicClient({
+  chain: mainnet,
+  transport: http(), // Uses default RPC for mainnet
+});
+
+interface EthPriceData {
+  price: number;
+  updatedAt: Date;
+}
+
+export interface UseGetEthPriceReturn {
+  data: EthPriceData | undefined;
+  isLoading: boolean;
+  isError: boolean;
+  error: Error | null;
+  refetch: () => void;
+}
+
+export const useGetEthPrice = (): UseGetEthPriceReturn => {
+  const { data, isLoading, isError, error, refetch } = useQuery<
+    EthPriceData,
+    Error
+  >({
+    queryKey: ["ethUsdPrice", ETH_USD_PRICE_FEED_ADDRESS, mainnet.id],
+    queryFn: async () => {
+      const [roundData, feedDecimals] = await Promise.all([
```

Comment:
getting the price from chainlink so that I can show an accurate fee in usd terms

Reply:
any reason to not use alchemy's token api?